components/GoalInput.js

```diff
@@ -0,0 +1,90 @@
+import {
+  StyleSheet,
+  View,
+  TextInput,
+  Button,
+  Text,
+  Modal,
+  Image,
+} from "react-native";
+import { useState } from "react";
+// ---------
+function GoalInput({ onAddGoal, visible, onCancel }) {
+  const [enteredText, setEnteredText] = useState(null);
+  const [empety, setEmpety] = useState(false);
```

Comment:
since it will have a bool value 
```suggestion
  const [isEmpety, setIsEmpety] = useState(false);
```